### ISSUE TITLE
No-op line continuation preceding comment.

### ISSUE THREAD

**Issue by cheako** [koalaman/shellcheck#954]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

echo yes \
# comment
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:

```
$ shellcheck myscript
Line 3:
echo yes \
         ^-- SC####: Remove \ at end of line, it does nothing.
```

**Comment by ColemanTom** [comment]
I would like to see this added too, it just caught me out.

**Comment by nicknovitski** [comment]
I had a related problem (really just from my own ignorance):

```bash
#!/usr/bin/env bash
pssh --hosts "$(host-inventory)" \
         --askpass \ # TODO: fix key authentication
         "$@"
```

ShellCheck reports no problem with this.  But of course, it doesn't work at all.  I thought `\` indicated a line continuation by itself, but it's actually _escaping the line break_, so if it isn't followed by a line break, but say, a space and a comment, then nothing works and everything is terrible.

I imagine something like:
```
$ shellcheck myscript
Line 2:
         --askpass \ # TODO: fix key authentication
                           ^-- SC####: Comment preceded by \, did you mean to escape a line break? 📎
```

I think that's just a generalization of this issue across the one-line and two-line cases.

**Comment by kevinoid** [comment]
I'd suggest also warning about a no-op line continuation following a comment.  I've run across code similar to the following:

```sh
echo \
        # Comment about option/argument \
        true
```

The backslash is discarded due to [shell tokenization rule 9](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_03), making the code equivalent to:

```sh
echo
        # Comment about option/argument
        true
```

Which is not likely to be what the author expected (whether they expected `true` to be part of the comment or an argument to `echo`).
